File: app/src/main/java/com/example/elderapp/SplashActivity.java
package com.example.elderapp;

import androidx.appcompat.app.AppCompatActivity;

import android.animation.Animator;
import android.animation.AnimatorListenerAdapter;
import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;
import android.util.Log;
import android.view.View;
import android.widget.ImageView;

public class SplashActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash);


        ImageView logo = findViewById(R.id.img_Logo);
        animateIn(logo);

        Handler handler = new Handler();
        handler.postDelayed(() -> {
            animateOut(logo);
            logo.setImageResource(R.drawable.ccu);
            animateIn(logo);
        }, 1500);

        handler.postDelayed(() -> {
            Intent it = new Intent(SplashActivity.this, LoginActivity.class);
            startActivity(it);
            finish();
        }, 3000);

//        Intent it = new Intent(SplashActivity.this, LoginActivity.class);
//        startActivity(it);
//        finish();

    }

    private void animateIn(View view){
        view.setAlpha(0f);
        view.setVisibility(View.VISIBLE);
        view.animate()
            .alpha(1f)
            .setDuration(1500)
            .setListener(null);



    }
    private void animateOut(View view){
        view.animate()
            .alpha(0f)
            .setDuration(1000)
            .setListener(new AnimatorListenerAdapter() {
                @Override
                public void onAnimationEnd(Animator animation) {
                    view.setVisibility(View.GONE);
                }
            });
    }
}

File: app/src/main/java/com/example/elderapp/SplashActivity.kt
package com.example.elderapp

import android.animation.Animator
import android.animation.AnimatorListenerAdapter
import android.content.Intent
import android.content.res.Configuration
import android.os.Bundle
import android.os.Handler
import android.view.View
import android.widget.ImageView
import androidx.appcompat.app.AppCompatActivity
import android.view.WindowManager
import android.util.Log






class SplashActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        adjustFontScale(resources.configuration);
        setContentView(R.layout.activity_splash)

        val logo = findViewById<ImageView?>(R.id.img_Logo)
        animateIn(logo)

        Handler().postDelayed({
            animateOut(logo)
            logo.setImageResource(R.drawable.ccu)
            animateIn(logo)
        }, 1500)
        Handler().postDelayed({
            val it = Intent(this@SplashActivity, LoginActivity::class.java)
            startActivity(it)
            finish()
        }, 3000)


    }

    private fun animateIn(view: View?) {
        view!!.alpha = 0f
        view.visibility = View.VISIBLE
        view.animate()
                .alpha(1f)
                .setDuration(1500)
                .setListener(null)
    }

    private fun animateOut(view: View?) {
        view!!.animate()
                .alpha(0f)
                .setDuration(1000)
                .setListener(object : AnimatorListenerAdapter() {
                    override fun onAnimationEnd(animation: Animator?) {
                        view.visibility = View.GONE
                    }
                })
    }

    private fun adjustFontScale(configuration: Configuration) {
        if (configuration.fontScale > 1.0) {
            Log.d("system",
                "fontScale=" + configuration.fontScale) //Custom Log class, you can use Log.w
            Log.d("system",
                "font too big. scale down...") //Custom Log class, you can use Log.w
            configuration.fontScale = 1.0f
            val metrics = resources.displayMetrics
            val wm = getSystemService(WINDOW_SERVICE) as WindowManager
            wm.defaultDisplay.getMetrics(metrics)
            metrics.scaledDensity = configuration.fontScale * metrics.density
            baseContext.resources.updateConfiguration(configuration, metrics)
        }
    }
}